### DISCUSSION TITLE
Facebook SDK Compatibility with iOS 15+ for App Store Uploads

### DISCUSSION THREAD

**Discussion by lukyanenk7** [codemagic-ci-cd/codemagic-docs#2464] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I am facing issues with asset validation when uploading my iOS app, built on Codemagic, to App Store Connect. The problem seems to stem from the Facebook SDK versions I am using, which are apparently not supporting the minimum OS version specified in my app’s Info.plist. My app targets iOS 15 and above, and the latest iOS version is 17.4.

Steps to reproduce:

1. Integrated Facebook SDK with version range set from 14.1.0 to less than 18.0.0 in my Swift Package Manager.
2. Configured the app to support a minimum iOS version of 15.0.
3. Attempted to upload to App Store Connect and received errors stating that the bundles for FBAEMKit.framework, FBSDKCoreKit.framework, and FBSDKCoreKit_Basics.framework do not support the minimum OS Version specified in the Info.plist.

### Expected results

Successful upload of the IPA to App Store Connect without compatibility issues.

### Actual results

Received errors from App Store Connect about the invalid bundle due to unsupported minimum OS version by the included Facebook SDK frameworks.

I would appreciate any insights into which version of the Facebook SDK I should be using to ensure compatibility with iOS 15+ and any steps I might need to follow to rectify this upload issue.

Thank you for your time and assistance.

### Build id (optional)

6617784cccbf228012845810

**Comment by lukyanenk7** [comment]
Found a solution, you need to manually set the latest version of fb specific, then skips publishing the application.

**Reply by icarusdust** [reply]
Hi @lukyanenk7, that would be a correct approach. The thing here is that Xcode 15.3 has a minimum OS version validation step that could potentially cause all the 3rd party dependencies to fail due to lower SDK versions being used. So, upgrading it to the latest was the correct approach you took, even though I did not quite get what you meant by skipping publishing when SDK is upgraded. Upgrading FB SDK to the latest should not skip publishing.

**Reply by naticio** [reply]
I have the same issue, I did not understand, just update the framework to the latest version?

**Comment by icarusdust** [comment]
Closing this thread due to inactivity, let me know if you have any questions by replying to this thread or starting a new one.
